vscode/src/globalState.ts

```diff
@@ -0,0 +1,151 @@
+/*
+  Copyright (c) 2023-2024, Oracle and/or its affiliates.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+import { TextEditorDecorationType, Uri } from 'vscode';
+import { NbTestAdapter } from './views/TestViewController';
+import { SetTextEditorDecorationParams } from './lsp/protocol';
+import { ExtensionContextInfo } from './extensionContextInfo';
+import { ClientPromise } from './lsp/clientPromise';
+import { NbProcessManager } from './lsp/nbProcessManager';
+
+export class GlobalState {
+    private static instance: GlobalState;
+    private initialized: boolean = false;
+    
+    private listeners: Map<string, string[]>;
+    private extensionContextInfo!: ExtensionContextInfo;
+    private clientPromise!: ClientPromise;
+    private debugPort: number;
+    private debugHash?: string;
+    // TODO: Find a way in nbcode.ts processOnCloseHandler to handle exit code, 
+    // so that deactivated can be removed from global state 
+    private deactivated: boolean;
+    private nbProcessManager: NbProcessManager | null;
+    private testAdapter?: NbTestAdapter;
+    private decorations: Map<string, TextEditorDecorationType>;
+    private decorationParamsByUri: Map<Uri, SetTextEditorDecorationParams>;
+
+    private constructor() {
+        this.listeners = new Map<string, string[]>();
+        this.debugPort = -1;
+        this.deactivated = true;
+        this.nbProcessManager = null;
+        this.decorations = new Map<string, TextEditorDecorationType>();
+        this.decorationParamsByUri = new Map<Uri, SetTextEditorDecorationParams>();
+    }
+
+    public static getInstance(): GlobalState {
```

Comment:
can we make this as  export const globalState = new GlobalState() and remove getInstance() method

Reply:
Done. Thanks!